kornia/augmentation/base.py

```diff
@@ -280,6 +280,81 @@ def identity_matrix(self, input) -> torch.Tensor:
         return kornia.eye_like(3, input)
 
 
+class IntensityAugmentationBase2D(AugmentationBase2D):
+    r"""IntensityAugmentationBase2D base class for customized intensity augmentation implementations.
+
+    For any augmentation, the implementation of "generate_parameters" and "apply_transform" are required while the
+    "compute_transformation" is only required when passing "return_transform" as True.
+
+    Args:
+        p (float): probability for applying an augmentation. This param controls the augmentation probabilities
+                   element-wisely for a batch.
+        p_batch (float): probability for applying an augmentation to a batch. This param controls the augmentation
+                         probabilities batch-wisely.
+        return_transform (bool): if ``True`` return the matrix describing the geometric transformation applied to each
+                                      input tensor. If ``False`` and the input is a tuple the applied transformation
+                                      wont be concatenated.
+        same_on_batch (bool): apply the same transformation across the batch. Default: False.
+        keepdim (bool): whether to keep the output shape the same as input (True) or broadcast it
+                        to the batch form (False). Default: False.
+    """
+
+    def compute_transformation(self, input: torch.Tensor, params: Dict[str, torch.Tensor]) -> torch.Tensor:
+        return self.identity_matrix(input)
+
+
+class GeometricAugmentationBase2D(AugmentationBase2D):
+    r"""GeometricAugmentationBase2D base class for customized geometric augmentation implementations.
+
+    For any augmentation, the implementation of "generate_parameters" and "apply_transform" are required while the
+    "compute_transformation" is only required when passing "return_transform" as True.
+
+    Args:
+        p (float): probability for applying an augmentation. This param controls the augmentation probabilities
+                   element-wisely for a batch.
+        p_batch (float): probability for applying an augmentation to a batch. This param controls the augmentation
+                         probabilities batch-wisely.
+        return_transform (bool): if ``True`` return the matrix describing the geometric transformation applied to each
+                                      input tensor. If ``False`` and the input is a tuple the applied transformation
+                                      wont be concatenated.
+        same_on_batch (bool): apply the same transformation across the batch. Default: False.
+        keepdim (bool): whether to keep the output shape the same as input (True) or broadcast it
+                        to the batch form (False). Default: False.
+    """
+
+    def inverse_transform(
```

Comment:
`compute_inverse_transform` ?

Reply:
this aims at the counterpart of ```apply_transform```. You have point that we can have the ```compute_inverse_transform ``` to get rid of the ````transform.inverse()```